src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -0,0 +1,310 @@
+
+    def __le__(self, other: Self) -> bool:
+        """Return whether this quantity is less than or equal to another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is less than or equal to another.
+        """
+        return self._base_value <= other._base_value
+
+    def __eq__(self, other: object) -> bool:
+        """Return whether this quantity is equal to another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is equal to another.
+        """
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return self._base_value == other._base_value
+
+
+class Power(
+    Quantity,
+    exponent_unit_map={
+        -3: "mW",
+        0: "W",
+        3: "kW",
+        6: "MW",
+    },
+):
+    """A power quantity."""
+
+    @classmethod
+    def from_watts(cls, watts: float) -> Self:
+        """Initialize a new power quantity.
+
+        Args:
+            watts: The power in watts.
+
+        Returns:
+            A new power quantity.
+        """
+        return cls(value=watts, exponent=0)
+
+    @classmethod
+    def from_milliwatts(cls, milliwatts: float) -> Self:
+        """Initialize a new power quantity.
+
+        Args:
+            milliwatts: The power in milliwatts.
+
+        Returns:
+            A new power quantity.
+        """
+        return cls(value=milliwatts, exponent=-3)
+
+    @classmethod
+    def from_kilowatts(cls, kilowatts: float) -> Self:
+        """Initialize a new power quantity.
+
+        Args:
+            kilowatts: The power in kilowatts.
+
+        Returns:
+            A new power quantity.
+        """
+        return cls(value=kilowatts, exponent=3)
+
+    @classmethod
+    def from_megawatts(cls, megawatts: float) -> Self:
+        """Initialize a new power quantity.
+
+        Args:
+            megawatts: The power in megawatts.
+
+        Returns:
+            A new power quantity.
+        """
+        return cls(value=megawatts, exponent=6)
+
+    def as_watts(self) -> float:
+        """Return the power in watts.
+
+        Returns:
+            The power in watts.
+        """
+        return self._base_value
+
+    def as_kilowatts(self) -> float:
+        """Return the power in kilowatts.
+
+        Returns:
+            The power in kilowatts.
+        """
+        return self._base_value / 1e3
+
+    def as_megawatts(self) -> float:
+        """Return the power in megawatts.
+
+        Returns:
+            The power in megawatts.
+        """
+        return self._base_value / 1e6
+
+    def __add__(self, other: Self) -> Self:
+        """Return the sum of this power and another.
+
+        Args:
+            other: The other power.
+
+        Returns:
+            The sum of this power and another.
+        """
+        return self.from_watts(self.as_watts() + other.as_watts())
```

Comment:
Should have used `_base_value` in the subclasses too,  there was a 10% perf difference, using the superclass constructor with _base_value.

But these can't be implemented in the superclass because not all quantities can get all operations.  For example voltages don't have addition.  And there might be other qtys with other requirements.

Reply:
Why can't voltage have addition?

Reply:
True,  it can have addition in serial connections,  and I guess they all can.  I was thinking only of parallel connections, for voltage. Will make it a common method.

Reply:
done